src/Aspire.Hosting/Health/ResourceHealthCheckService.cs

```diff
@@ -93,12 +93,8 @@ await resourceNotificationService.PublishUpdateAsync(resource, s =>
                 {
                     var healthReports = MergeHealthReports(s.HealthReports, report);
 
-                    // Matches the logic in ASP.NET Core's private HealthReport.CalculateAggregateStatus
-                    var healthStatus = healthReports.MinBy(r => r.Status)?.Status ?? s.HealthStatus;
-
                     return s with
                     {
-                        HealthStatus = healthStatus,
                         HealthReports = healthReports
```

Comment:
nit: Add a comment that the health status on the snapshot is automatically re-calculated from health reports.